src/services/NodeMonitor.ts

```diff
@@ -298,7 +306,8 @@ export class NodeMonitor {
 			) {
 				return;
 			}
-			const nodeInx = this.nodeList.findIndex((addedNode) => addedNode.publicKey === node.publicKey);
+
+			const nodeInx = this.nodeList.findIndex((addedNode) => addedNode.publicKey === node.publicKey || addedNode.host === node.host);
```

Comment:
Why do you need to check the hostname also?

Reply:
I added it to check the hostname because I found (sometimes) the list from the peer list has duplicate hostname with different public keys.

When the duplicated hostname is in the list, the query `node/info` will return node info with the same public key.

As the public key in node tables (DB) is `unique` it will cause an insert fail.

But you can ignore this, I will revert to commit to this. I will remove duplicated data before insert to DB.

Reply:
Can you give an example of a node that does this?

Reply:
The node have the same host name, with different public key.

host: shoestring.pasomi.net

https://xym85.allnodes.me:3001/node/peers
https://xym139.allnodes.me:3001/node/peers

Reply:
It seems like that issue will be recovered or fixed by the network.
I can't find the same host name with different public key any more.

I will revert this change.

Reply:
Yeah, the network should update once the new node information syncs.